src/azul/service/avro_pfb.py

```diff
@@ -143,7 +154,8 @@ def from_json(cls,
         # For files, document_id is not unique (because of related_files), but
         # uuid is.
         ids = [object_['uuid']] if name == 'files' else sorted(object_['document_id'])
-        return cls(id='_'.join((name, *ids)), name=name, object=object_)
+        id_ = str(uuid5(cls.namespace_uuid, '_'.join((name, *ids))))
```

Comment:
Assert that joiner doesn't occur in input.

```suggestion
        id_ = f"{name}_{uuid5(cls.namespace_uuid, '_'.join(ids))}_{len(ids}}"
```

Reply:
> Assert that joiner doesn't occur in input.
If I apply this to `name` too, then my diff looks different from your suggestion, so I'm intentionally diverging.